binary_transparency/firmware/devices/usb_armory/README.md

```diff
@@ -6,25 +6,99 @@ In this package is support for using the
 
 Storage
 -------
 
 > :warning: these are scratch notes, not yet reflective of reality, and so may
 > change drastically!
 
-The _plan_ is to partition the device SD card into 3:
-1. `Boot` partition, this will contain our "fake ROM" which does the proof verification
-3. `Bundle` partition, this will contain the proof bundle
-4. `Firmware` partition, this contains the logged firmware image to boot if
-    verification succeeds.
+We'll use the µSD card slot of the USB Armory for our purposes.
+
+The SD card will contain:
+   - our "enforcing" [bootloader](./bootloader)
+   - the firmware being made discoverable
+   - a [proof bundle](/binary_transparency/firmware/api/update_package.go)
+     for the firmware which convinces the bootloader that it _is_ discoverable
+     and therefore ok to launch.
+
+> :info: the USB Armory is built around an NXP i.MX6 SoC.  When booting, the ROM
+> loader on this SoC expects to find the first-stage bootloader at the
+> 1024th byte of the external storage.
+> This allows sufficient space beforehand to store a partition table.
+
+The on-disk partition layout will be:
+index | name       | size    | format | notes
+------|------------|---------|--------|-----------------------------------------------
+1     | boot       | 10M     | raw    | Must cover disk bytes 1024 onwards as we'll directly write the bootloader here.
+2     | proof      | 512KB   | ext4   | EXT4 filesytem for storing a serialised proof bundle
+3     | firmware   | 64MB+   | ext4   | EXT4 filesystem containing the bootable firmware  image, armory boot config, and DTD file.
+
+### Preparing the SD Card
+
+> :warning: When following the instructions below, be *very sure* you know which
+> device is your SD card - if performed with an incorrect device, the instructions below
+> can cause data loss!
+
+#### Linux
+
+##### Partition & file-systems
+
+First use the `parted -l` command to figure out which device corresponds to your
+SD card.
+> :tip: you can run the `parted -l` command twice, once with your SD card
+> reader plugged in, and once without to help identify the device.
+
+`/dev/my_sdcard` is used as a placeholder below, you should replace that with
+the path for your SD card device.
+
+```bash
+sudo parted /dev/my_sdcard
+# double (triple!) check we've got the right device:
+(parted) print
+...
+(parted) mklabel msdos
+# Create space for the bootloader
+(parted) mkpart primary 1KB 10240KB
+# Create a partition for the proofs
+(parted) mkpart primary ext4 10241KB 10753KB
+# Create a partition for the firmwware
+(parted) mkpart primary ext4 10754KB 100MB
+
+# Check our work:
+(parted) unit b
+(parted) print
+Model: Generic- Micro SD/M2 (scsi)
+Disk /dev/sdc: 15931539456B
+Sector size (logical/physical): 512B/512B
+Partition Table: msdos
+Disk Flags: 
 
-Compiling
----------
+Number  Start      End         Size       Type     File system  Flags
+ 1      512B       10240511B   10240000B  primary               lba
```

Comment:
Maybe a case of the Mondays, but I'm confused why this say 512B which is not equal to 1024B as stated above?

Reply:
Yeah, I was surprised by that too, but that's what came out... it doesn't strictly matter at the mo' since we're `dd`'ing the bootloader to exactly where we want to it be (@1024B), and that location is covered by this partition entry anyway, but it's a bit displeasing.